Documentation/Contributing/UnitTests.md

```diff
@@ -77,55 +76,107 @@ To create a new test the following template can be used:
+            // NOTE: This setup initializes the main camera at Vector3(1.0f, 1.5f, -2.0f)
+            PlayModeTestUtilities.Setup();
         }
 
-        // do common setup for each of your tests here - this will be called for each individual test after entering playmode
-        [SetUp]
-        public void Init()
+        // Destroy commonly initialized objects here - this will be called after each of your tests has finished
+        [TearDown]
+        public virtual void TearDown()
         {
-            // in most play mode test cases you would want to at least create an MRTK GameObject using the default profile
-            TestUtilities.InitializeMixedRealityToolkit(true);
+            PlayModeTestUtilities.TearDown();
         }
+    }
+}
+#endif
 
+```
 
-        // destroy commonly initialized objects here - this will be called after each of your tests has finished
-        [TearDown]
-        public void Shutdown()
+## Creating a new play mode test
+To create a new play mode test:
+- Navigate to Assets > MixedRealityToolkit.Tests > PlayModeTests
+- Right click, Create > Testing > C# Test Script
+- Replace the default template with the example starter test below
+
+``` csharp
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// The assembly that this file is part of is still built for the player,
+// even though the assembly itself is marked as a test assembly (this is not
+// expected because test assemblies should not be included in player builds).
+// Because the .NET backend is deprecated in 2018 and removed in 2019 and this
+// issue will likely persist for 2018, this issue is worked around by wrapping all
+// play mode tests in this check.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using NUnit.Framework;
+using System;
+using System.Collections;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    class ExamplePlayModeTests : BasePlayModeTests
+    {
+        // Override Setup() in BasePlayModeTests to change the starting camera position
+        public override void Setup()
         {
-            // call shutdown if you've created an mrtk GameObject in your test
-            TestUtilities.ShutdownMixedRealityToolkit();
+            base.Setup();
+            
+            // Set camera position to Vector3(0,0,0), the default position Vector3(1.0f, 1.5f. -2.0f)
+            TestUtilities.PlayspaceToOriginLookingForward();
         }
 
-
         #region Tests
 
+        /// <summary>
+        /// Example starter test that shows a test hand, gets the line pointer and moves the hand
+        /// </summary>
         [UnityTest]
-        /// the name of this method will be used as test name in the unity test runner
-        public IEnumerator TestMyFeature()
+        public IEnumerator ExampleTest()
```

Comment:
I think what's missing that this change tries to address is the following:

I am a person who is writing a playmode test for the first time, and I don't even know what is available to use. If I see a completely empty skeleton with just the "EmptyTestFunction()". What are the capabilities that I can use? What are the functions I can invoke?

This is a definite gap in the stuff down below, because ideally the test infrastructure that we  have would also have some docs (i.e. a section in our docs about our testing infra along with APIs you can use like "how to simulate hands in tests" or "how to simulate HL1 gaze in tests").

Without that test infra documented, I can see how wanting to have a basic example here would fill that gap.

That being said, I agree with Julia that having a skeleton + having an implementation of some unrelated tests are two different things.

Here are some things I think that we could do:

1. Mention in bold that the skeleton below is more than a skeleton, and that you don't need to actually have any of the stuff in the test.
2. Possibly have the code in the fake test commented out along with additional comments about when you might want to comment out the following lines (i.e. a "build your test" sort of situation)
3. Delete the code inside the fake test and actually add the docs for the testing infra.

I think 1 and 2 might be good?

Reply:
I decided to go the fake test as a comment route (2).  Here is what I currently have for the test section of the skeleton.

```

        #region Tests

        /// <summary>
        /// Skeleton for a new MRTK play mode test.
        /// </summary>
        [UnityTest]
        public IEnumerator TestMyFeature()
        {
            // ----------------------------------------------------------
            // EXAMPLE SIMPLE HAND INPUT TEST
            // ----------------------------------------------------------
            // Getting the input system
            // var inputSystem = PlayModeTestUtilities.GetInputSystem();

            // Creating a new test hand for input
            // var rightHand = new TestHand(Handedness.Right);
            // yield return rightHand.Show(new Vector3(0, 0, 0.5f));

            // Moving the new test hand
            // We are doing a yield return here because moving the hand to a new position
            // requires multiple frames to complete the action.
            // yield return rightHand.MoveTo(new Vector3(0, 0, 2.0f));

            // Getting a specific pointer from the hand
            // var linePointer = PointerUtils.GetPointer<LinePointer>(Handedness.Right);
            // Assert.IsNotNull(linePointer);
            // ---------------------------------------------------------

            // Your new test here
            yield return null;
        }
        #endregion
```
When the comment is removed, it becomes the skeleton but how to at least create and move the hand is there.  Also I think having documentation addressing how to create specific types of tests would be really useful.

@wiwei 
What do you think of something like this for now?

Reply:
lgtm